### PR TITLE
Authentication method can now be customized

### PR DESCRIPTION
**Pull request by majamassarini**
Fixes packit/packit-service#1879

Merge before packit/packit-service#1902

- [x] Add unit tests

### CODE REVIEW

File: ogr/services/github/service.py [hunk 1]
```diff
@@ -86,16 +88,32 @@ def __init__(
...
+    def reset_auth_method(self):
+        logger.info("Reset Github auth method to the default")
+        self._other_auth_method = None
+
+    @property
+    def authentication(self):
+        if self._other_auth_method:
+            return self._other_auth_method
+        else:
+            return self._default_auth_method
```
Comment (lbarcziova) [inline ogr/services/github/service.py:116]:
```suggestion
        return self._other_auth_method or self._default_auth_method
```

**Review by lbarcziova — APPROVED** [review]
:+1:

**Comment by mfocko** [comment]
regate

**Comment by mfocko** [comment]
what the hell is wrong with you /o\

**Comment by mfocko** [comment]
Never mind, I can't read apparently… :/

**Pull request merged**